wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 2 - Stale Message

Parent #15625

This PR `displays a stale message below the site switcher` whenever (and only when) there's a dashboard update error. Unlike the dashboard error snack bar, it keeps displaying until the update is successful. 

UI 

Light Mode | Dark Mode
------------| -----------
![stale_msg_light_mode](https://user-images.githubusercontent.com/1405144/147342386-cb21928a-576c-4763-a8c3-ab1825cf98d2.png) | ![stale_msg_dark_mode](https://user-images.githubusercontent.com/1405144/147342372-ee843048-b315-480b-9902-856ae7f6237e.png)

To test:

Prerequisites

- Go to `My Site` -> `Me` -> `App Settings` -> `Test Feature Configuration`.
- Turn the `MySiteDashboardPhase2FeatureConfig` feature flag ON.
- Relaunch the app.

1. Login with an account having `wpcom` sites and go to the `My Site` tab.
2. Notice that initially the stale message is not displayed below the site switcher.
3. Update the `CardsSource.getData(...)` and `CardsSource.fetchCardsAndPostErrorIfAvailable(...)` logic to get the `showStaleMessage` to true.
4. Re-install the app.
5. Notice that the stale message is displayed below the site switcher.

## Regression Notes
1. Potential unintended areas of impact
The stale message should not be displayed on login from `self-hosted` site or inside the `Jetpack` app.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
See above.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the stale message display and order. 

PR submission checklist:
